src/NuGet.Core/NuGet.ProjectModel/Utf8JsonStreamReader.cs

```diff
@@ -54,10 +55,19 @@ internal Utf8JsonStreamReader(Stream stream, int bufferSize = BufferSizeDefault,
             _buffer = _bufferPool.Rent(bufferSize);
             _disposed = false;
             _stream = stream;
-#pragma warning disable CA2022 // Avoid inexact read
-            _stream.Read(_buffer, 0, 3);
-#pragma warning restore CA2022
-            if (!Utf8Bom.AsSpan().SequenceEqual(_buffer.AsSpan(0, 3)))
+            try
```

Comment:
I think this can be simplified.  We don't need to handle a new exception or introduce an extension method.

```C#
if (_stream.Read(_buffer, offset: 0, count: 1) == 1 &&
    _stream.Read(_buffer, offset: ++_bufferUsed, count: 1) == 1 &&
    _stream.Read(_buffer, offset: ++_bufferUsed, count: 1) == 1)
{
    ++_bufferUsed;

    bool hasUtf8Bom = Utf8Bom.AsSpan().SequenceEqual(_buffer.AsSpan(start: 0, length: 3));

    if (hasUtf8Bom)
    {
        _bufferUsed = 0;
    }
}
```

Reply:
Update: https://github.com/NuGet/NuGet.Client/pull/6350/commits/6ec62bdb9f9aabe638f5c9e6d27522af029b20be